### PR TITLE
refactor/entitlements-broadcast

### PR DESCRIPTION
**Pull request by dsshap**
- adding broadcast-specific functions
- updating tests
- adding script for setting role/capabilities

### CODE REVIEW

File: src/core/RolesAuthority.sol [hunk 1]
```diff
@@ -131,7 +127,7 @@ contract RolesAuthority is IAuthority, Initializable, UUPSUpgradeable {
     }
 
     function setRoleCapability(Role role, address target, bytes4 functionSig, bool enabled) public virtual {
-        role == Role.System_FundAdmin ? _assertOwner() : _assertPermissions();
+        uint8(role) > uint8(Role.Investor_Reserve5) ? _assertOwner() : _assertPermissions();
```
Comment (dsshap) [inline src/core/RolesAuthority.sol:130]: @swissarmytowel in the lambda ensure that the right wallet is being used

File: test/fuzz/RolesAuthority.t.sol
Comment (dsshap) [inline test/fuzz/RolesAuthority.t.sol:1]: @swissarmytowel can you add some fuzz testers for owner

File: src/core/RolesAuthority.sol (hunk 1 above)
Comment (dsshap) [reply src/core/RolesAuthority.sol]: updated but still relevent

File: test/fuzz/RolesAuthority.t.sol
Comment (dsshap) [reply test/fuzz/RolesAuthority.t.sol:1]: updated but still relevent

**Review by swissarmytowel — APPROVED** [review]

**Pull request merged**